node/src/transition_frontier/transition_frontier_effects.rs

```diff
@@ -307,10 +307,18 @@ fn synced_effects<S: crate::Service>(
             best_tip: best_tip.block.clone(),
         });
     }
-    // rebroadcast block if received from webrtc network, otherwise noop.
-    store.dispatch(P2pNetworkPubsubAction::WebRtcRebroadcast {
-        message: GossipNetMessageV2::NewState(best_tip.block().clone()),
-    });
+    // TODO this should be handled by a callback
+    // If this get dispatched, we received block from libp2p.
+    if !store.dispatch(P2pNetworkPubsubAction::BroadcastValidatedMessage {
+        message_id: p2p::BroadcastMessageId::BlockHash {
+            hash: best_tip.hash().clone(),
+        },
+    }) {
+        // Otherwise block was received from WebRTC so inject it in libp2p.
+        store.dispatch(P2pNetworkPubsubAction::WebRtcRebroadcast {
```

Comment:
what would happen if we dispatched this one unconditionally? is the conditional check what avoids the duplication of the message? (I ask because we can do this here because the transition frontier still doesn't use queued actions, but once that is implemented we will have to figure out something else here)

Reply:
>is the conditional check what avoids the duplication of the message?

>but once that is implemented we will have to figure out something else here)

+ to both